### ISSUE TITLE
Enhance activations testing

### ISSUE THREAD

**Issue by jalvesz** [fortran-lang/stdlib#992]
### Motivation

A module with activation function was added in #860 

It was proposed to include the means to check the functions against known libraries such as PyTorch or TensorFlow. A way to do that would be to add a Python script that evalues the available activations and their derivatives and stores them as a list of `.npy` files. These files should then be read in the tests to compare the results

**Comment by ashshiv123** [comment]
Dear @jalvesz,

Thank you for the guidance. As you advised, I am posting this initial draft here for discussion. The proposed Python script is designed to be placed within the `test/specialfunctions` folder, allowing anyone to regenerate the reference data as needed.

My draft focuses on the `sigmoid` function as a starting point.

---

### **Python Script for Reference Data Generation**

This script generates the reference input, output, and derivative values for the `sigmoid` function using PyTorch and saves them as `.npy` files.

```python
import numpy as np
import torch
import os

# Define the name of the function
activation_name = 'sigmoid'

# Use float64 for precision consistent with Fortran's real(dp)
x_np = np.linspace(-10.0, 10.0, num=201, dtype=np.float64)

# Convert to a PyTorch tensor to track gradients
x_torch = torch.tensor(x_np, requires_grad=True)

# Calculate the sigmoid function output
y_torch = torch.sigmoid(x_torch)

# Calculate the derivative
y_torch.sum().backward()
dydx_np = x_torch.grad.numpy()

# Save results to .npy files
np.save(f'{activation_name}_input.npy', x_np)
np.save(f'{activation_name}_output.npy', y_torch.detach().numpy())
np.save(f'{activation_name}_derivative.npy', dydx_np)

print(f"Reference data for '{activation_name}' has been generated successfully.")

**Comment by ashshiv123** [comment]
Hi @jalvesz,

I hope you are doing well. I wanted to kindly follow up on the initial draft I shared above for the activation testing script (focused on the sigmoid function).

Could you please review it and let me know if the approach and placement (inside test/specialfunctions) align with the project’s expectations? I’d be happy to refine the script further, extend it to other activation functions, and then open a PR once it looks good to you.

Thank you for your time and guidance!

Best regards,
@ashshiv123

**Comment by jalvesz** [comment]
Hi @ashshiv123 thanks for looking at this and sorry for the late reply.

On the python side it would be nice that the script is structured starting with the list of activation functions currently available in stdlib. Being at the level of the test/specialfunctions folder, it should generate a "data" folder at that level and within write all the files.

Next step would be on the Fortran side to use this data for the comparison.

**Comment by learner0018** [comment]
Thank you @jalvesz, I see now. I will integrate this into the existing fypp test suite using test-drive and ensure stdlib_io is linked properly
